pkg/containers/containers.go

```diff
@@ -182,62 +202,98 @@ func (c *Containers) populate() error {
 // saving container information in Containers CgroupInfo map.
 // NOTE: ALL given cgroup dir paths are stored in CgroupInfo map.
 func (c *Containers) CgroupUpdate(cgroupId uint64, path string, ctime time.Time) (CgroupInfo, error) {
-	containerId, runtime := getContainerIdFromCgroup(path)
+	containerId, containerRuntime := getContainerIdFromCgroup(path)
+	container := cruntime.ContainerMetadata{
+		ContainerId: containerId,
+	}
 
 	info := CgroupInfo{
-		Path:        path,
-		ContainerId: containerId,
-		Runtime:     runtime,
-		Ctime:       ctime,
+		Path:      path,
+		Container: container,
+		Runtime:   containerRuntime,
+		Ctime:     ctime,
 	}
 
 	c.mtx.Lock()
 	c.cgroups[uint32(cgroupId)] = info
 	c.mtx.Unlock()
 
+	err := c.enrichCgroupInfo(cgroupId)
+
+	//don't fail on enrichment but log to stderr
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "failed to enrich container data: %v", err)
+	}
+
 	return info, nil
 }
 
+// enrichCgroupInfo checks for a given cgroupId if it is relevant to some running container
+// it then calls the runtime info service to gather additional data from the container's runtime
+func (c *Containers) enrichCgroupInfo(cgroupId uint64) error {
+	c.mtx.RLock()
+	info := c.cgroups[uint32(cgroupId)]
+	c.mtx.RUnlock()
+	containerId := info.Container.ContainerId
```

Comment:
You make an assumption here that info was initialized, which is true now, but might change in the future for whatever reason.

Reply:
Fixed this assumption to return early with an error if the cgroup doesn't exist anymore.